docs/src/api/class-browsercontext.md

```diff
@@ -119,6 +131,15 @@ page = page_info.value
 print(page.evaluate("location.href"))
 ```
 
+```csharp
+var popupTask = context.WaitForPageAsync();
+await Task.WhenAll(popupTask, page.ClickAsync("a"))
+    .WithTimeout(1000) // add using Microsoft.Playwright.Helpers; to get this helper
```

Comment:
It's intentionally not the same. As you point out, the idea here is to have a way to make sure the example does not dead-lock if something goes wrong _downstream_ from us. And this also applies to both tasks and the potential opportunity for a deadlock. Granted, in this example, that possibility is low, but as a general pattern, waiting for both tasks to complete might introduce a dead-lock and I think the right thing to do, is have a good pattern for it everywhere we use it.

Reply:
What kind of deadlock are you talking about? In the present system, we only manage timeouts on the server side, so I'm with Yury here and I actually left a similar comment in another example. We should only use timeouts inside ClickAsync and WaitForPageAsync.

Reply:
I think you've got a point - the "example" code comes from our (.net) tests so it's a bit more defensive (i.e. we don't want it hanging in case we made a coding mistake and mis-handled something from downstream) which is where that timeout comes into play. 

So, from a simplicity perspective, I'm happy to loose it. We can add a section on best practices later/in places more appropriate. Considering this isn't _test_ or _library_ code per se, we can get rid of the `ConfigureAwait` as well.

Reply:
Yeah, let's remove the `WithTimeout`. That's internal. Let's also remove the `ConfigureAwait` bellow.